nautobot/ipam/factory.py

```diff
@@ -111,3 +164,381 @@ def prefix(self, n):
+        None,
+    )
+
+    @factory.post_generation
+    def children(self, create, extracted, **kwargs):
+        """
+        Create child prefixes and ip addresses within the prefix IP space. Defaults to generating
+        0-4 child prefixes for parents that are containers, or generating 0-4 ip addresses within
+        the prefix's IP space for non-container prefixes. Only creates children on 50% of prefixes.
+        Set children__max_count to an integer when calling the factory creation methods
+        (`create()`, `create_batch()`, etc) to override the maximum number of children generated.
+        """
+        if extracted:
+            # Objects have already been created, do nothing
+            return
+
+        # 50% chance to create children
+        if not faker.Faker().pybool():
+            return
+
+        action = "create" if create else "build"
+        is_ipv6 = self.family == 6
+
+        # Create child prefixes for containers, otherwise create child ip addresses
+        factory = PrefixFactory if self.status == Prefix.STATUS_CONTAINER else IPAddressFactory
+        method = getattr(factory, action)
+
+        # Default to maximum of 4 children unless overridden in kwargs
+        max_count = int(kwargs.pop("max_count", 4))
+        child_count = faker.Faker().pyint(min_value=0, max_value=min(max_count, self.prefix.size))
+        if child_count == 0:
+            return
+
+        if factory == IPAddressFactory:
+            # Create child ip addresses, preserving vrf, tenant and is_ipv6 from parent
+            for count, address in enumerate(self.prefix.iter_hosts()):
+                if count == child_count:
+                    break
+                method(address=str(address), vrf=self.vrf, tenant=self.tenant, is_ipv6=is_ipv6, **kwargs)
+        else:
+            # Calculate prefix length for child prefixes to allow them to fit in the parent prefix without creating duplicate prefix
+            child_cidr = self.prefix_length + max(1, math.ceil(math.log(child_count, 2)))
+
+            # Raise exception for invalid cidr length (>128 for ipv6, >32 for ipv4)
+            if child_cidr > 128 or self.family == 4 and child_cidr > 32:
+                raise ValueError(f"Unable to create {child_count} child prefixes in container prefix {self.cidr_str}.")
+
+            # Create child prefixes, preserving tenant, site, location, vrf and is_ipv6 from parent
+            for count, address in enumerate(self.prefix.subnet(child_cidr)):
+                if count == child_count:
+                    break
+                method(
+                    prefix=str(address.cidr),
+                    tenant=self.tenant,
+                    site=self.site,
+                    location=self.location,
+                    is_container=False,
+                    is_ipv6=is_ipv6,
+                    vrf=self.vrf,
+                    **kwargs,
+                )
+
+
+class IPAddressFactory(PrimaryModelFactory):
```

Comment:
Seems like we'll want a docstring for this factory as well in keeping with `AggregateFactory` and `PrefixFactory`?